src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/RazorDocumentExcerptService.cs

```diff
@@ -53,83 +47,46 @@ public RazorDocumentExcerptService(DocumentSnapshot document, IRazorSpanMappingS
                 return null;
             }
 
-            var mapped = await _mapper.MapSpansAsync(document, new[] { span }, cancellationToken).ConfigureAwait(false);
+            var mapped = await _mappingService.MapSpansAsync(document, new[] { span }, cancellationToken).ConfigureAwait(false);
             if (mapped.Length == 0 || mapped[0].Equals(default(RazorMappedSpanResult)))
             {
                 return null;
             }
 
             var project = _document.Project;
-            var primaryDocument = project.GetDocument(mapped[0].FilePath);
-            if (primaryDocument == null)
+            var razorDocument = project.GetDocument(mapped[0].FilePath);
```

Comment:
Interesting that we never seem to use any thing but the first element, is there a reason we need to be returning more? Maybe we could just stop when we find the first one inside MapSpansAsync. Also not a fan of the name "mapped", maybe `mappedSpans`?

Reply:
> is there a reason we need to be returning more
We just return 1, however I'm following the existing `IRazorSpanMappingService` interface which specifies a `ImmutableArray<RazorMappedSpanResult>` instead of a single result. This is within the ExternalAccess package and I don't believe is easily modifiable unfortunately / may have external dependencies.

> Maybe we could just stop when we find the first one inside MapSpansAsync
`MapSpansAsync` provides a 1:1 mapping so returning excess results / excess processing isn't an issue.

> Also not a fan of the name "mapped", maybe mappedSpans?
Sure thing!